alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt

```diff
@@ -109,6 +111,41 @@ class TransportGetFindingsSearchAction @Inject constructor(
             queryBuilder.filter(QueryBuilders.termsQuery("monitor_id", getFindingsRequest.monitorIds))
         }
 
+        if (!detectionType.isNullOrBlank()) {
+            val nestedQueryBuilder = QueryBuilders.nestedQuery(
+                "queries",
+                when {
+                    detectionType.equals("threat", ignoreCase = true) -> {
+                        QueryBuilders.boolQuery().filter(
+                            QueryBuilders.prefixQuery("queries.id", "threat_intel_")
+                        )
+                    }
+                    else -> {
+                        QueryBuilders.boolQuery().mustNot(
+                            QueryBuilders.prefixQuery("queries.id", "threat_intel_")
```

Comment:
What's the reason to filter out the threat intel queries here? As a user I would expect the filter to be based on the provided `detectionType`

Reply:
Yes, I have categorized the detectionType parameter into two types: "threat" and normal rule based.  If the frontend API calls the findings API with detectionType set to "threat," all threat intel-related findings will be generated with this query. I'm open to considering alternative solutions for this categorization.

Reply:
I think the enum comment I had on the other CR makes this a bit more clear. Having the arbitrary text input + looking for exactly 2 cases seems like it should be enum-based.

If we don't expect the detectionType options to expand, then having a boolean field seems more user-friendly to me. Something like `getThreatIntelFindingsOnly` (name could use some work) that allows the user to pick which type of findings are returned.

Or maybe a new API, say GetThreatIntelFindings, would also make sense to me.

Another thought on this - this change is arguably a breaking one. We are changing the default behavior of an existing API to now filter out the threat intel findings. If a user has a system that relies on this API already and they are expecting threat intel findings, then it would break their use case.

Is there a way we can explore doing the filtering on the frontend, or at least leaving the default behavior as-is? I'm a bit concerned with merging this at the moment, but open to others' thoughts.

Reply:
Missed line 128.. not a breaking change or change to default behavior since filtering is only applied if a detectionType is supplied